contracts/src/integrations/cert/EigenDACertVerifier.sol

```diff
@@ -46,22 +72,76 @@ contract EigenDACertVerifier is
         if (initSecurityThresholds.confirmationThreshold <= initSecurityThresholds.adversaryThreshold) {
             revert InvalidSecurityThresholds();
         }
+        if (initQuorumNumbersRequired.length == 0 || initQuorumNumbersRequired.length > 256) {
+            revert InvalidQuorumNumbersRequired(initQuorumNumbersRequired.length);
+        }
         _eigenDAThresholdRegistry = initEigenDAThresholdRegistry;
         _eigenDASignatureVerifier = initEigenDASignatureVerifier;
         _securityThresholds = initSecurityThresholds;
         _quorumNumbersRequired = initQuorumNumbersRequired;
     }
 
+    /// @notice Decodes a certificate from bytes to an EigenDACertV3
+    /// @dev This function is external for the purpose of try/catch'ing it inside checkDACert.
+    function decodeCert(bytes calldata data) external pure returns (CT.EigenDACertV3 memory cert) {
+        return abi.decode(data, (CT.EigenDACertV3));
+    }
+
     /// @inheritdoc IEigenDACertVerifierBase
+    /// @dev This function try catches checkDACertReverts, and maps any reverts to status codes.
+    /// @dev Make sure to call this at a block number that is > RBN, otherwise this function will
+    /// return an INVALID_CERT status code because of a require in the BLSSignatureChecker library that we use.
+    /// TODO: we should return (uint8, bytes) instead and include the revert reason.
```

Comment:
should we file a linear ticket for this?

Reply:
https://linear.app/eigenlabs/issue/DAINT-733/checkdacert-should-return-bytes-reason-along-with-status-code